modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/function/ListSizeFunction.java

```diff
@@ -0,0 +1,53 @@
+package org.geotools.mbstyle.function;
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+
+import java.util.List;
+
+/**
+ * Returns the size of a given list
+ */
+public class ListSizeFunction extends FunctionExpressionImpl {
+
+    public static FunctionName NAME = new FunctionNameImpl("listSize",
+            parameter("size", Integer.class),
+            parameter("list", List.class));
```

Comment:
You could support Collection here, rather then merely List (List is a subclass of Collection). Wouldn't make a difference for the current functionality, but could be useful in the future.